source/PlayerInfo.cpp

```diff
@@ -1039,13 +1039,9 @@ void PlayerInfo::SetFlagship(Ship &other)
 	// Set the new flagship pointer.
 	flagship = other.shared_from_this();
 
-	// Make sure your jump-capable ships all know who the flagship is.
+	// Make sure your ships all know who the flagship is.
 	for(const shared_ptr<Ship> &ship : ships)
-	{
-		bool shouldFollowFlagship = (ship != flagship && !ship->IsParked() &&
-			(!ship->CanBeCarried() || ship->JumpNavigation().JumpFuel()));
-		ship->SetParent(shouldFollowFlagship ? flagship : shared_ptr<Ship>());
-	}
+		ship->SetParent(flagship);
```

Comment:
The flagships and ships that are parked should probably still have an empty pointer passed into this method.
Passing the flagship pointer into everything results in parked ships and the falgship itself being added to the flagship's escort list, which might cause undesirable side effects.